File: app/src/main/java/com/andy/logdog/MainActivity.java
package com.andy.logdog;

import android.os.Bundle;
import android.util.Log;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.io.File;

public class MainActivity extends AppCompatActivity {
    EditText editText;
    private TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        File file = new File(getApplicationContext()
                .getExternalFilesDir(null), "logdog");
        Logdog.path = file.getAbsolutePath();

        editText = findViewById(R.id.et);
        tv = findViewById(R.id.tv);
        // Example of a call to a native method
        findViewById(R.id.btn_read).setOnClickListener(v-> readFile());
        findViewById(R.id.btn_write_mmap).setOnClickListener(v-> writeInputWithMMAP());
        findViewById(R.id.btn_init).setOnClickListener(v-> Logdog.getInstance().init(getApplication()));
    }

    private void writeInputWithMMAP() {
        Logdog.getInstance().w(getInput());
        editText.setText("");
    }

    private String getInput() {
        String input;
        if (editText.getText().length() <= 0){
           input = "";
        } else {
           input = editText.getText().toString();
        }
        return input;
    }

    private void readFile() {
        String read = Logdog.getInstance().readFile(Logdog.path);
        Log.i("Main", "read from file: " + read);
        tv.setText(read);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Logdog.getInstance().onExit();
    }
}

File: app/src/main/java/com/andy/logdog/MainActivity.kt
package com.andy.logdog

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        Logdog.getInstance().i("MainActivity.onCreate invoked")
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Example of a call to a native method
        btn_read.setOnClickListener { readFile() }
        btn_write_mmap.setOnClickListener { writeInputWithMMAP() }
    }

    private fun getEditInput(): String {
        if(et.text.isEmpty()) return ""
        return et.text.toString()
    }

    private fun writeInputWithMMAP() {
        Logdog.getInstance().w(getEditInput())
        et.setText("")
    }

    private fun readFile() {
        val read = Logdog.getInstance().read()
        Log.i("Main", "read from file: $read")
        tv.text = read
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        Logdog.getInstance().d("MainActivity.onAttachedToWindow invoked")
    }

    override fun onStart() {
        super.onStart()
        Logdog.getInstance().d("MainActivity.onStart invoked")
    }

    override fun onResume() {
        super.onResume()
        Logdog.getInstance().d("MainActivity.onResume invoked")
    }

    override fun onPause() {
        super.onPause()
        Logdog.getInstance().d("MainActivity.onPause invoked")
    }

    override fun onStop() {
        super.onStop()
        Logdog.getInstance().d("MainActivity.onStop invoked")
    }

    override fun onDestroy() {
        super.onDestroy()
        Logdog.getInstance().i("MainActivity.onDestroy invoked")
    }
}
